### ISSUE TITLE
Bug - Route not drawn on ND

### ISSUE THREAD

**Issue by jelovac** [legoboyvdlp/A320-family#158]
**Describe the bug**

Sometimes imported flight plan fails to be displayed on ECAM display.

Looking at the console in Windows it seems to be a nasal error.

**To Reproduce**
Steps to reproduce the behaviour:
1. Initialize the MCDU
2. Load external flightplan via FMGC => Load flightplan

Flightplan:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<PropertyList>
	<source type="string">https://flightplandatabase.com/plan/3108288</source>
	<version type="int">2</version>
	<departure>
		<airport type="string">LFPG</airport>
	</departure>
	<destination>
		<airport type="string">EGBB</airport>
	</destination>
	<route>
		<wp n="0">
			<type type="string">navaid</type>
			<departure type="bool">true</departure>
			<alt-restrict type="string">at</alt-restrict>
			<altitude-ft type="double">392</altitude-ft>
			<ident type="string">LFPG</ident>
			<lat type="double">49.009722</lat>
			<lon type="double">2.547778</lon>
		</wp>
		<wp n="1">
			<type type="string">navaid</type>
			<alt-restrict type="string">at</alt-restrict>
			<altitude-ft type="double">9000</altitude-ft>
			<ident type="string">PG092</ident>
			<lat type="double">49.031333</lat>
			<lon type="double">2.715028</lon>
		</wp>
		<wp n="2">
			<type type="string">navaid</type>
			<alt-restrict type="string">at</alt-restrict>
			<altitude-ft type="double">25500</altitude-ft>
			<ident type="string">OPALE</ident>
			<lat type="double">49.899750</lat>
			<lon type="double">1.885083</lon>
		</wp>
		<wp n="3">
			<type type="string">navaid</type>
			<alt-restrict type="string">at</alt-restrict>
			<altitude-ft type="double">26000</altitude-ft>
			<ident type="string">KESAX</ident>
			<lat type="double">50.070833</lat>
			<lon type="double">1.674167</lon>
		</wp>
		<wp n="4">
			<type type="string">navaid</type>
			<alt-restrict type="string">at</alt-restrict>
			<altitude-ft type="double">26000</altitude-ft>
			<ident type="string">DIMAL</ident>
			<lat type="double">50.492500</lat>
			<lon type="double">1.366389</lon>
		</wp>
		<wp n="5">
			<type type="string">navaid</type>
			<alt-restrict type="string">at</alt-restrict>
			<altitude-ft type="double">26000</altitude-ft>
			<ident type="string">ALESO</ident>
			<lat type="double">50.575556</lat>
			<lon type="double">1.225556</lon>
		</wp>
		<wp n="6">
			<type type="string">navaid</type>
			<alt-restrict type="string">at</alt-restrict>
			<altitude-ft type="double">26000</altitude-ft>
			<ident type="string">ROTNO</ident>
			<lat type="double">50.857222</lat>
			<lon type="double">0.778056</lon>
		</wp>
		<wp n="7">
			<type type="string">navaid</type>
			<alt-restrict type="string">at</alt-restrict>
			<altitude-ft type="double">26000</altitude-ft>
			<ident type="string">ETVAX</ident>
			<lat type="double">50.968608</lat>
			<lon type="double">0.598964</lon>
		</wp>
		<wp n="8">
			<type type="string">navaid</type>
			<alt-restrict type="string">at</alt-restrict>
			<altitude-ft type="double">26000</altitude-ft>
			<ident type="string">TIGER</ident>
			<lat type="double">51.067222</lat>
			<lon type="double">0.439444</lon>
		</wp>
		<wp n="9">
			<type type="string">navaid</type>
			<alt-restrict type="string">at</alt-restrict>
			<altitude-ft type="double">26000</altitude-ft>
			<ident type="string">BIG</ident>
			<lat type="double">51.330875</lat>
			<lon type="double">0.034811</lon>
		</wp>
		<wp n="10">
			<type type="string">navaid</type>
			<alt-restrict type="string">at</alt-restrict>
			<altitude-ft type="double">26000</altitude-ft>
			<ident type="string">ULTIB</ident>
			<lat type="double">51.660000</lat>
			<lon type="double">-0.278889</lon>
		</wp>
		<wp n="11">
			<type type="string">navaid</type>
			<alt-restrict type="string">at</alt-restrict>
			<altitude-ft type="double">26000</altitude-ft>
			<ident type="string">HEMEL</ident>
			<lat type="double">51.805556</lat>
			<lon type="double">-0.419444</lon>
		</wp>
		<wp n="12">
			<type type="string">navaid</type>
			<alt-restrict type="string">at</alt-restrict>
			<altitude-ft type="double">26000</altitude-ft>
			<ident type="string">BUZAD</ident>
			<lat type="double">51.942222</lat>
			<lon type="double">-0.552222</lon>
		</wp>
		<wp n="13">
			<type type="string">navaid</type>
			<alt-restrict type="string">at</alt-restrict>
			<altitude-ft type="double">26000</altitude-ft>
			<ident type="string">OLNEY</ident>
			<lat type="double">52.127778</lat>
			<lon type="double">-0.734167</lon>
		</wp>
		<wp n="14">
			<type type="string">navaid</type>
			<alt-restrict type="string">at</alt-restrict>
			<altitude-ft type="double">26000</altitude-ft>
			<ident type="string">WELIN</ident>
			<lat type="double">52.247222</lat>
			<lon type="double">-0.852222</lon>
		</wp>
		<wp n="15">
			<type type="string">navaid</type>
			<alt-restrict type="string">at</alt-restrict>
			<altitude-ft type="double">18700</altitude-ft>
			<ident type="string">D103L</ident>
			<lat type="double">52.314425</lat>
			<lon type="double">-1.345058</lon>
		</wp>
		<wp n="16">
			<type type="string">navaid</type>
			<alt-restrict type="string">at</alt-restrict>
			<altitude-ft type="double">13200</altitude-ft>
			<ident type="string">HON</ident>
			<lat type="double">52.356678</lat>
			<lon type="double">-1.663725</lon>
		</wp>
		<wp n="17">
			<type type="string">navaid</type>
			<alt-restrict type="string">at</alt-restrict>
			<altitude-ft type="double">6400</altitude-ft>
			<ident type="string">OSKOT</ident>
			<lat type="double">52.341667</lat>
			<lon type="double">-2.058333</lon>
		</wp>
		<wp n="18">
			<type type="string">navaid</type>
			<alt-restrict type="string">at</alt-restrict>
			<altitude-ft type="double">3600</altitude-ft>
			<ident type="string">GROVE</ident>
			<lat type="double">52.393889</lat>
			<lon type="double">-1.928889</lon>
		</wp>
		<wp n="19">
			<type type="string">navaid</type>
			<arrival type="bool">true</arrival>
			<alt-restrict type="string">at</alt-restrict>
			<altitude-ft type="double">339</altitude-ft>
			<ident type="string">EGBB</ident>
			<lat type="double">52.453889</lat>
			<lon type="double">-1.748056</lon>
		</wp>
	</route>
</PropertyList>
```

**Expected behaviour**
Route should be displayed on ECAM display.

**Screenshots**

![nasal-a320-ecam-route-bug](https://user-images.githubusercontent.com/3627139/85204895-f5f03000-b317-11ea-85d7-27781d87d15e.png)

![a320-ecam-fp-bug](https://user-images.githubusercontent.com/3627139/85204882-e375f680-b317-11ea-9dd6-5832c29f1354.gif)

** System (please complete the following information):**
 - OS: Windows 10
 - FlightGear version: 2020.1.2

**Comment by legoboyvdlp** [comment]
yeah, if that happens you have to live with it (or restart flightgear).

Can you reproduce it -- if not, unable to fix.

**Comment by merspieler** [comment]
That's the mapstructure issue again... i thought it was fixed?

**Comment by jelovac** [comment]
> yeah, if that happens you have to live with it (or restart flightgear).
> 
> Can you reproduce it -- if not, unable to fix.

You are correct, restarting Flightgear and importing the same flight plan works.

As it happens randomly not sure if I can find the exact pattern in order to reproduce it.

**Comment by legoboyvdlp** [comment]
It's obviously not, merspieler.

The problem is that the debug information is clearly not working... it's meant to output exactly which model it can't find but desn't...

**Comment by merspieler** [comment]
I've got the very same today... entering a flightplan manually....
`YPDN/11 RUSKA5 RUSKA DCT DUMAV J61 KONDA CODIE7X YBCS/15`

**Comment by legoboyvdlp** [comment]
@jelovac 

@merspieler 

I wonder if you could both change your log level to debug -- OR else on line 24 of mapstructure.nas, change it like so:

#var _MP_dbg_lvl = LOG_DEBUG;
var _MP_dbg_lvl = DEV_ALERT;


After that, you should see in the logfile something like "Deleting symbol of type X"

**Comment by jelovac** [comment]
> 
> 
> @jelovac
> 
> @merspieler
> 
> I wonder if you could both change your log level to debug -- OR else on line 24 of mapstructure.nas, change it like so:
> 
> #var _MP_dbg_lvl = LOG_DEBUG;
> var _MP_dbg_lvl = DEV_ALERT;
> 
> After that, you should see in the logfile something like "Deleting symbol of type X"

Ok, will do it and let you know the outcome when map fails.

**Comment by jelovac** [comment]
Not sure if it is the same bug but I discovered the following:

- Start Flightgear
- Connect external power
- Set departure and arrival airport in the MCDU
- Load flightplan

All good.

Then from Flightgear menu move your aircraft to some other location, for example spawn it on the active runway => the map is gone.

Some infos:

```
In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-fbw.xml: line 8
      Property fbw/roll/pid is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-fbw.xml: line 9
      Property fbw/pitch/g-pi is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-fbw.xml: line 10
      Property fbw/pitch/pid is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-fbw.xml: line 11
      Property fbw/yaw/tc-pd is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-fbw.xml: line 12
      Property fbw/spoiler-output is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-fbw.xml: line 13
      Property fbw/mixer/sidestick-gain-feedback is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-fbw.xml: line 16
      Property fbw/roll/p-gain is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-fbw.xml: line 17
      Property fbw/roll/i-gain is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-fbw.xml: line 18
      Property fbw/roll/d-gain is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-fbw.xml: line 19
      Property fbw/pitch/p-gain is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-fbw.xml: line 20
      Property fbw/pitch/i-gain is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-fbw.xml: line 21
      Property fbw/pitch/d-gain is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-fbw.xml: line 22
      Property fbw/yaw/yd-p-gain is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-fbw.xml: line 23
      Property fbw/yaw/tc-p-gain is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-fbw.xml: line 24
      Property fbw/yaw/tc-d-gain is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-fbw.xml: line 27
      Property fbw/fmgc/roll-cmd is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-fbw.xml: line 28
      Property fbw/fmgc/pitch-cmd is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-fbw.xml: line 29
      Property fbw/fmgc/yaw-cmd is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-fcs.xml: line 8
      Property /controls/flight/flaps-input-last is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-spoiler.xml: line 7
      Property hydraulics/spoiler-l2/main-cmd-feedback is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-electrical.xml: line 7
      Property /systems/electrical/some-electric-thingie/rat-deploy is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-hydraulic.xml: line 7
      Property /systems/hydraulic/sys-incr-rate is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-aerodynamics.xml: line 7
      Property ice/wing is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-aerodynamics.xml: line 8
      Property rubbish/landL is already defined.

In file D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Systems/a320-aerodynamics.xml: line 9
      Property rubbish/landR is already defined.
```

And then the following nasal error:

```
  503.93 [ALRT]:nasal      Nasal runtime error: non-objects have no members
  503.93 [ALRT]:nasal        at D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Nasal/FMGC/flightplan.nas, line 826
  503.93 [ALRT]:nasal        called from: D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Nasal/FMGC/flightplan.nas, line 760
  503.93 [ALRT]:nasal        called from: D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Nasal/FMGC/flightplan.nas, line 64
  503.93 [ALRT]:nasal        called from: D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Nasal/Libraries/libraries.nas, line 228
  503.94 [ALRT]:nasal        called from: D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Nasal/Libraries/libraries.nas, line 233
  503.94 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/globals.nas, line 119
  503.94 [ALRT]:nasal      Loading saved state...
  503.94 [ALRT]:nasal      Saved state loaded
  512.27 [ALRT]:flight     Failed to save initial FDM property state
  512.67 [ALRT]:nasal      Nasal runtime error: non-objects have no members
  512.67 [ALRT]:nasal        at D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Nasal/FMGC/flightplan.nas, line 826
  512.76 [ALRT]:nasal      Nasal runtime error: non-objects have no members
  512.76 [ALRT]:nasal        at D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Nasal/FMGC/flightplan.nas, line 826
  512.84 [ALRT]:nasal      Nasal runtime error: non-objects have no members
  512.84 [ALRT]:nasal        at D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Nasal/FMGC/flightplan.nas, line 826
  512.94 [ALRT]:nasal      Nasal runtime error: non-objects have no members
```

The last three line keeps repeating forever.

**Comment by jelovac** [comment]
Test 2

Started on a parking spot W05 on LFPG

1. Enabled BAT 1 & 2
2. Enabled APU
3. Inserted arrival and departure airports into the MCDU via keyboard input by clicking Toggle Keyboard Input
4. Moved with the pushback truck the aircraft to the another location
5. Set the parking brake
6. Loaded flight plan from file (the one reported LFPG - EGBB)

Outcome: route not displayed, nasal error in console.

Reproduced twice in a row.

```
 508.60 [ALRT]:nasal      Nasal runtime error: function/method call on uncallable object
  508.60 [ALRT]:nasal        at C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 38
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 440
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 884
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1077
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1104
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/geo.nas, line 448
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1051
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1059
  508.60 [ALRT]:nasal        called from: D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Models/Instruments/ND/canvas/map/RTE.lcontroller, line 29
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/globals.nas, line 119
  508.60 [ALRT]:nasal      Nasal runtime error: MapStructure model not found
  508.60 [ALRT]:nasal        at C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 27
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1104
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/geo.nas, line 448
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1051
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1059
  508.60 [ALRT]:nasal        called from: D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Models/Instruments/ND/canvas/framework/canvas.nas, line 245
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/globals.nas, line 119
  508.60 [ALRT]:nasal      Nasal runtime error: function/method call on uncallable object
  508.60 [ALRT]:nasal        at C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 38
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 440
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 727
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1077
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1104
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/geo.nas, line 448
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1051
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1059
  508.60 [ALRT]:nasal        called from: D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Models/Instruments/ND/canvas/framework/canvas.nas, line 245
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/globals.nas, line 119
  508.60 [ALRT]:nasal      Nasal runtime error: function/method call on uncallable object
  508.60 [ALRT]:nasal        at C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 38
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 440
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 884
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1077
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1104
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/geo.nas, line 448
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1051
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1059
  508.60 [ALRT]:nasal        called from: D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Models/Instruments/ND/canvas/map/RTE.lcontroller, line 29
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/globals.nas, line 119
  508.60 [ALRT]:nasal      Nasal runtime error: MapStructure model not found
  508.60 [ALRT]:nasal        at C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 27
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1104
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/geo.nas, line 448
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1051
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1059
  508.60 [ALRT]:nasal        called from: D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Models/Instruments/ND/canvas/framework/canvas.nas, line 245
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/globals.nas, line 119
  508.60 [ALRT]:nasal      Nasal runtime error: function/method call on uncallable object
  508.60 [ALRT]:nasal        at C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 38
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 440
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 727
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1077
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1104
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/geo.nas, line 448
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1051
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1059
  508.60 [ALRT]:nasal        called from: D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Models/Instruments/ND/canvas/framework/canvas.nas, line 245
  508.60 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/globals.nas, line 119
  508.61 [ALRT]:nasal      Nasal runtime error: MapStructure model not found
  508.61 [ALRT]:nasal        at C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 27
  508.61 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1104
  508.61 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/geo.nas, line 448
  508.61 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1051
  508.61 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1059
  508.61 [ALRT]:nasal        called from: D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Models/Instruments/ND/canvas/map/RTE.lcontroller, line 37
  508.61 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/globals.nas, line 119
  508.61 [ALRT]:nasal      Nasal runtime error: MapStructure model not found
  508.61 [ALRT]:nasal        at C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 27
  508.61 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1104
  508.61 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/geo.nas, line 448
  508.61 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1051
  508.61 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1059
  508.61 [ALRT]:nasal        called from: D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Models/Instruments/ND/canvas/map/RTE.lcontroller, line 37
  508.61 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/globals.nas, line 119
  508.61 [ALRT]:nasal      Nasal runtime error: MapStructure model not found
  508.61 [ALRT]:nasal        at C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 27
  508.61 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1104
  508.61 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/geo.nas, line 448
  508.61 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1051
  508.61 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1059
  508.61 [ALRT]:nasal        called from: D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Models/Instruments/ND/canvas/map/RTE.lcontroller, line 37
  508.61 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/globals.nas, line 119
  508.61 [ALRT]:nasal      Nasal runtime error: MapStructure model not found
  508.61 [ALRT]:nasal        at C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 27
  508.61 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1104
  508.61 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/geo.nas, line 448
  508.61 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1051
  508.61 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/canvas/MapStructure.nas, line 1059
  508.61 [ALRT]:nasal        called from: D:/Users/vladar/Documents/FlightGear/Custom Aircraft/A320-family/Models/Instruments/ND/canvas/map/RTE.lcontroller, line 37
  508.61 [ALRT]:nasal        called from: C:/Program Files/FlightGear 2020.1.2/data/Nasal/globals.nas, line 119
```

**Comment by legoboyvdlp** [comment]
Moving the aircraft is not supported. Will see if it is possible to fix flightplan bug.

But, did you set log level to debug -- you should have seen "Deleting symbol of type {something}"

Edit - please can you try and get it, but without moving aircraft location between airports?

**Comment by jelovac** [comment]
> But, did you set log level to debug -- you should have seen "Deleting symbol of type {something}"

Yes I did, that is all which I got in the Windows terminal output.

> Edit - please can you try and get it, but without moving aircraft location between airports?

Second test was on the same airport, but with moving the aircraft like 30m from the starting point with the pushback.

I will keep an eye and try to log the bug if it occurs when aircraft is not moved at all before FP is set.

**Comment by jelovac** [comment]
> > But, did you set log level to debug -- you should have seen "Deleting symbol of type {something}"
> 
> Yes I did, that is all which I got in the Windows terminal output.

Scratch this. 

Apparently in Windows version of FGFS 2020.1.2 it was by default set to:

var _MP_dbg_lvl = LOG_DEBUG;
#var _MP_dbg_lvl = DEV_ALERT;

I just swapped the values and unfortunately tested with DEV_ALERT. I will repeat the tests after Wednesday.

**Comment by legoboyvdlp** [comment]
I see -- using pushback is no problem :) 

Ok, not a problem, either swapping values _or_ setting --log-level=debug is fine!

Actually for DEV_ALERT -- looks like you have to add --developer=yes - very sorry, I forgot this! So for both, you have to add something to commandline.

**Comment by legoboyvdlp** [comment]
By the way,

It appears you need to use the console (--console or else "Show Debugging Console" in settings on the launcher) to see log level DEBUG messages - not entirely sure why that is. 

It appears DEBUG messages are not output to fgfs.log. I will open a ticket for this on the mailing list tonight!

**Comment by legoboyvdlp** [comment]
Should be fixed by 5e666793
